### PR TITLE
DS-5170 Make consent optional

### PR DESCRIPTION
**Pull request by OH17L**
## HTT
- [x] When you log in as X then you should see the following message **We published a new version of the data policy. You can review the data policy here.**
- [x] When you click **here** in that message then you should be on the **Data policy agreement** page
- [x] You should see that the **I agree with the data policy** checkbox is optional
- [x] Press **Save** and you should be redirected to the homepage
- [x] You should not see the following message **We published a new version of the data policy. You can review the data policy here.**
- [x] Log in as administrator and go to **/admin/config/people/data-policy/1**
- [x] You should see one row in the **USER CONSENTS FOR CURRENT REVISION** table and you should see **No** in the **Agreed** column
- [x] Go to **/admin/structure/data_policy/settings**
- [x] Set **Enforce consent** checkbox and press **Save configuration**
- [x] When you log in as X again then you should be on the **Data policy agreement** page
- [x] You should see that the **I agree with the data policy** checkbox is required
- [x] Set **I agree with the data policy** checkbox and press **Save**
- [x] Log in as administrator and go to **/admin/config/people/data-policy/1**
- [x] You should see two row in the **USER CONSENTS FOR CURRENT REVISION** table and you should see **Yes** in the **Agreed** column of a new row

### CODE REVIEW

File: src/RedirectSubscriber.php [hunk 1]
```diff
@@ -69,17 +93,62 @@ public static function getSubscribedEvents() {
...
+      ->loadByProperties($values);
+
+    if (!empty($user_consents)) {
+      return;
+    }
+
+    if (!$enforce_consent) {
+      $link = Link::createFromRoute(t('here'), 'gdpr_consent.data_policy.agreement');
+
+      \Drupal::messenger()->addStatus(t('You can agree with the data policy @url.', [
```
Comment (bramtenhove) [inline src/RedirectSubscriber.php]: We should this text a little bit, maybe something like:

> We published a new revision of the data policy. You can review the data policy here.

**Comment by OH17L** [comment]
@bramtenhove, when user declined optional consent then creating new user consent entity and for **status** field of this entity setting **FALSE** value.

File: src/RedirectSubscriber.php (hunk 1 above)
Comment (OH17L) [reply src/RedirectSubscriber.php]: @bramtenhove, can I use word **version** instead of **revision**? I think it will be more clear for users.

**Comment by bramtenhove** [comment]
@OH17L, I see. But when a user is **undecided** then we also need to store this.

Given the following situation:

1. Consent is optional and thus not enforced
2. Site manager creates new revision
3. User X logs in and message about revision is shown
4. User X navigates to data policy agreement page
5. User X just navigates away (doesn't save the form)

In this case the user is **undecided** and we should store this action as it may be required later on for the SM. It's _not_ the same as not agreeing to the data policy as the user did not explicitly do this.

**Pull request merged**